autoload/maktaba/extension.vim

```diff
@@ -0,0 +1,159 @@
+""
+" @dict ExtensionRegistry
+" A registry for the extensions used a single plugin.  Extensions are
+" dictionaries with a plugin-specific interpretation
```

Comment:
When I see "dictionary", I often forget it can be something more like an "object" that can have function fields. It might make it easier to understand its purpose to call out that it can contain functions and other data.

Also including inline code examples here or pointing to the examples in vroom would go a long way, since it's a pretty abstract feature.